crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -750,6 +733,92 @@ impl DecisionTree {
         }
     }
 
+    // returns condition1 AND condition2
+    fn and_conditions(
+        condition1: Option<NodeId>,
+        condition2: Option<NodeId>,
+        stack_frame: &mut StackFrame,
+        ctx: &mut SsaContext,
+    ) -> Option<NodeId> {
+        match (condition1, condition2) {
+            (None, None) => None,
+            (Some(condition), None) | (None, Some(condition)) => {
+                if condition.is_dummy() {
+                    None
+                } else {
+                    Some(condition)
+                }
+            }
+            (Some(cond1), Some(cond2)) => {
+                if cond1.is_dummy() {
+                    Self::and_conditions(None, condition2, stack_frame, ctx)
+                } else if cond2.is_dummy() {
+                    Self::and_conditions(None, condition1, stack_frame, ctx)
+                } else if cond1 == cond2 {
+                    condition1
+                } else {
+                    let op = Operation::Binary(node::Binary {
+                        lhs: cond1,
+                        rhs: cond2,
+                        operator: BinaryOp::Mul,
+                        predicate: None,
+                    });
+                    let cond = ctx.add_instruction(Instruction::new(
+                        op,
+                        ObjectType::Boolean,
+                        Some(stack_frame.block),
+                    ));
+                    optimizations::simplify_id(ctx, cond).unwrap();
+                    stack_frame.push(cond);
+                    Some(cond)
+                }
+            }
+        }
+    }
+
+    // returns condition1 OR condition2
+    fn or_conditions(
+        condition1: Option<NodeId>,
+        condition2: Option<NodeId>,
+        stack_frame: &mut StackFrame,
+        ctx: &mut SsaContext,
+    ) -> Option<NodeId> {
+        match (condition1, condition2) {
+            (_condition, None) | (None, _condition) => None,
```

Comment:
Why do we return None here instead of the other condition? It somewhat reads like AND and OR functions are switched

Reply:
None is the same as true in this context.